### DISCUSSION TITLE
[Feature request] Straight z-seam painting

### DISCUSSION THREAD

**Discussion by bionik** [OrcaSlicer/OrcaSlicer#3441] [Ideas]
I would like to suggest a feature request for z-seam painting. 

It's sometimes hard do draw the z-seam in straight line. In Photoshop there is a way to paint straight lines by first clicking the starting point and with shift pressed you click another point. Photoshop then draws a straight line to the ending point. 

This would not be too hard to implement in Orca. You would need to keep track of the last point and then when clicking to the new point when shift is pressed, you would interpolate (in screen cordinates) a line between the points. This could be implemented as synthetic clicks in small increments between these coordinates.

**Comment by Fusiable** [comment]
Within the painting option, just below the sliders, there is a check box for that feature called "Vertical" 
It makes seam painting SO much better.

**Reply by bionik** [reply]
That is convenient, but not all models have vertical corners. The straight line painting option would be very beneficial for these situations.

**Reply by SnijderC** [reply]
Alternatively, using the outline of the manually drawn line, then finding a median value that represents a straight line, something like this:

<img width="641" alt="screenshot" src="https://github.com/user-attachments/assets/846b9b8b-5dc6-4b4f-8642-74b6e5ae2aa1" />

Or just allow to create a modifier, like adding a cylinder, within which the seam must be rendered, the radius can then decide how strictly it's followed if there are reasons to optimize within the radius at all.

**Comment by dannypike** [comment]
I've just started using seam painting in Orca slicer and I'm a bit surprised that this feature isn't implemented, yet. My distinctly "average" motor control means that doing this by mouse is not accurate enough to make the feature even worth having. So I don't understand the point of being allowed to paint a seam "vaguely", for non-artists like me. :)

**Comment by LJKraus** [comment]
AI produces the instructions faster than I can:

> While OrcaSlicer does not have an official, dedicated "straight line" tool for all painting functions, a community-found trick and specific settings can achieve this in Seam Painting and Color Painting. 
>
> **Seam Painting Straight Lines**
> A specific key-stroke sequence can be used to draw a straight seam line between two points in any direction: 
> Enter seam painting mode by pressing P (or clicking the Seam Painting tool).
> Click and hold the mouse button at your desired start point.
> While still holding the mouse button down, tap P again to exit seam painting mode.
> Release the mouse button.
> Move your cursor to the desired end point of the line.
> Tap P once more to re-enter seam painting mode.
> Click the mouse button (down and up quickly) at the end point to complete the straight line between the two locations. 
> For purely vertical seams, an easier method is to use the built-in "Vertical" option for seam placement in the Seam Settings, which automatically aligns the seam vertically on the model.

**Reply by bionik** [reply]
Nice workaround, but I honestly think you should do the similar thing with pressing shift while you click. Seems like it should be quite a straight forward way to implement that since a workaround like this works.

**Comment by dannypike** [comment]
That's perfect, thank you!

(When I asked Anthropic Claude, all that I got was "you can't, look at this forum and vote for the 'why hasn't it been done yet' idea". I decided to write my own report as well. It looks like I need more practice with my prompts, as well as everything else with this CAM hobby I've just started.)

**Comment by Ataraxia-Mechanica** [comment]
Adding support for snapping seams to the nearest edge should help with some situations as well, it's a thing in cura slicer
<img width="909" height="1520" alt="image" src="https://github.com/user-attachments/assets/61cae28e-1a3a-438f-80c7-df567853189f" />
